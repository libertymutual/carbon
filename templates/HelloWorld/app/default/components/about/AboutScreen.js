import React from 'react';
import PropTypes from 'prop-types';
import { Button, ScrollView, StyleSheet, Text } from 'react-native';
import { connect } from 'react-redux';

import * as PresentationConstants from '../../constants/PresentationConstants';

/**
 * About Screen
 * @extends {React.Component}
 */
export class AboutScreen extends React.Component {
  /**
   * Renders the components
   * @override
   * @returns {JSX}
   */
  render() {
    return (<ScrollView>
      <Text style={styles.aboutScreenHeader}>{PresentationConstants.ABOUT_WELCOME}</Text>
      <Text style={styles.aboutScreenText}>{PresentationConstants.ABOUT_INTRO}</Text>
      <Button
        onPress={this._navigateBack}
        title={PresentationConstants.GO_BACK}
      />
    </ScrollView>);
  }

  /**
   * Navigates back to the previous screen boop
   * @private
   */
  _navigateBack = () => {
    this.props.navigation.goBack();
  }
}

/**
 * Navigation Options
 */
AboutScreen.navigationOptions = {
  title: PresentationConstants.PAGE_TITLE_ABOUT_SCREEN,
};

/**
 * Prop Types
 */
AboutScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

/**
 * Maps State To Props
 * @returns {Object) Object representing the parts of state
 * that the component should subscribe to
 */
function mapStateToProps() {
  return {};
}

/**
 * Maps Dispatch Actions To Props
 * @returns {Object) Object representing the actions
  * that should be exposed to the component via props
 */
function mapDispatchToProps() {
  return {};
}

/**
 * Default export wrapped in a connect to expose state & actions via props
 */
export default connect(mapStateToProps, mapDispatchToProps)(AboutScreen);

/**
 * Component Styles
 */
const styles = StyleSheet.create({
  aboutScreenHeader: {
    margin: 14,
    fontWeight: 'bold',
  },
  aboutScreenText: {
    margin: 14,
  },
});
