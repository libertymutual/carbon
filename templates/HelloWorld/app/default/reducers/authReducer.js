import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
  isLoggedIn: false,
  username: "",
  password: "",
  authUrl: "",
  clientId: "",
  clientSecret: "",
  configName: "",
  validatorId: "",
}

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function authReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.LOGIN:
      return { ...state, isLoggedIn: true };
    case ActionTypes.LOGOUT:
      return { ...state, isLoggedIn: false };
    case ActionTypes.AUTH_UPDATE_AUTH_URL:
      return { ...state, authUrl: action.authUrl };
    case ActionTypes.AUTH_UPDATE_CLIENT_ID:
      return { ...state, clientId: action.clientId };
    case ActionTypes.AUTH_UPDATE_CLIENT_SECRET:
      return { ...state, clientSecret: action.clientSecret };
    case ActionTypes.AUTH_UPDATE_CONFIG_NAME:
      return { ...state, configName: action.configName };
    case ActionTypes.AUTH_UPDATE_PASSWORD:
      return { ...state, password: action.password };
    case ActionTypes.AUTH_UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.AUTH_UPDATE_VALIDATOR_ID:
      return { ...state, validatorId: action.validatorId };
    default:
      return state;
  }
}
