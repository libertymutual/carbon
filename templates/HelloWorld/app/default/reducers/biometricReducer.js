import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
    username: "TestUsername",
    password: "TestPassword",
    credentialsSaved: false
};

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function biometricReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.UPDATE_PASSWORD:
      return { ...state, password: action.password };  
    case ActionTypes.SAVE_CREDENTIALS:
      return { ...state, credentialsSaved: true };
    case ActionTypes.DELETE_CREDENTIALS:
      return { ...state, credentialsSaved: false };
    default:
      return state;
  }
}
