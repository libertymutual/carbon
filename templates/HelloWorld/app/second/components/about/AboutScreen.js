import React from 'react';
import PropTypes from 'prop-types';
import { Button, ScrollView, StyleSheet, Text } from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as Constants from '../../constants/Constants';
import * as PresentationConstants from '../../constants/PresentationConstants';
import * as StylingConstants from '../../constants/StylingConstants';
import HelloWorldActions from '../../actions/HelloWorldActions';

/**
 * About Screen
 * @extends {React.Component}
 */
class AboutScreen extends React.Component {

  /**
   * Renders the components
   * @override
   * @returns {JSX}
   */
  render() {
    let welcomeMessage;

    if (this.props.helloWorld.state === Constants.HELLO) {
      welcomeMessage = PresentationConstants.ABOUT_HELLO_WORLD;
    } else {
      welcomeMessage = PresentationConstants.ABOUT_GOODBYE_WORLD;
    }

    return (<ScrollView style={styles.secondFeatureTheme}>
      <Text style={styles.aboutScreenHeader}>{PresentationConstants.ABOUT_WELCOME}</Text>
      <Text style={styles.aboutScreenText}>{welcomeMessage}</Text>
      <Button
        onPress={this._toggleWelcomeMessage}
        title={PresentationConstants.ABOUT_BUTTON_LABEL_TOGGLE}
      />
      <Button
        onPress={this._navigateBack}
        title={PresentationConstants.GO_BACK}
      />
    </ScrollView>);
  }

  /**
   * Toggles the welcome message
   * @private
   */
  _toggleWelcomeMessage = () =>  {
    if (this.props.helloWorld.state === Constants.HELLO) {
      this.props.helloWorldActions.goodbye();
    } else {
      this.props.helloWorldActions.hello();
    }
  }

  /**
   * Navigates back to the previous screen
   * @private
   */
  _navigateBack = () => {
    this.props.navigation.goBack();
  }
}

/**
 * Navigation Options
 */
AboutScreen.navigationOptions = {
  title: PresentationConstants.PAGE_TITLE_ABOUT_SCREEN,
};

/**
 * Prop Types
 */
AboutScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  helloWorld: PropTypes.object.isRequired,
  helloWorldActions: PropTypes.object.isRequired,
};

/**
 * Maps State To Props
 * @param {Object} state Redux State Object
 * @returns {Object) Object representing the parts of state
 * that the component should subscribe to
 */
function mapStateToProps(state) {
  return {
    helloWorld: state.helloWorld,
  };
}

/**
 * Maps Dispatch Actions To Props
 * @param {function} Dispatch function
 * @returns {Object) Object representing the actions
 * that should be exposed to the component via props
 */
function mapDispatchToProps(dispatch) {
  return {
    helloWorldActions: bindActionCreators(HelloWorldActions, dispatch),
  };
}

/**
 * Default export wrapped in a connect to expose state & actions via props
 */
export default connect(mapStateToProps, mapDispatchToProps)(AboutScreen);

/**
 * Component Styles
 */
const styles = StyleSheet.create({
  aboutScreenHeader: {
    margin: 14,
    fontWeight: 'bold',
  },
  aboutScreenText: {
    margin: 14,
  },
  secondFeatureTheme: {
    backgroundColor: StylingConstants.COLOR_LIGHTBLUE,
  },
});
