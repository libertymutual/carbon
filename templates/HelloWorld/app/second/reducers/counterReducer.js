import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = 0;

/**
 * Reducer handling counter related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function countReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.INCREMENT:
      return state + 1;
    case ActionTypes.DECREMENT:
      return state - 1;
    default:
      return state;

  }
}
