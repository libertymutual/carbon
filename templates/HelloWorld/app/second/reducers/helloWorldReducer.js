import * as ActionTypes from '../actions/ActionTypes';
import * as Constants from '../constants/Constants';

/**
 * Initial Reducer State
 */
export const initialState = {
  state: Constants.HELLO,
};

/**
 * Reducer handling hello world related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function helloWorldReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.HELLO:
      return { ...state, state: Constants.HELLO };
    case ActionTypes.GOODBYE:
      return { ...state, state: Constants.GOODBYE };
    default:
      return state;
  }
}
