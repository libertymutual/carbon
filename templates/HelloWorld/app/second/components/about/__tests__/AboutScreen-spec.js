import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import AboutScreen from '../AboutScreen';
import { initialState } from '../../../reducers/helloWorldReducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
let store;

describe('AboutScreen tests', () => {
  beforeEach(() => {
    // Sets up a new store
    store = mockStore({
      helloWorld: initialState
    });
  });

  test('About Screen renders correctly', () => {
    const tree = renderer.create(
      <AboutScreen store={store} navigation={{}}/>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('Goodbye message displays when helloWorld state is hello', () => {

    // Override the default initialState so state is goodbye
    store = mockStore({
      helloWorld: {
        state: 'goodbye'
      }
    });
    const tree = renderer.create(
      <AboutScreen store={store} navigation={{}}/>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
