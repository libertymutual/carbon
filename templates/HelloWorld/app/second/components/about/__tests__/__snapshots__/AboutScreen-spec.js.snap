// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen tests About Screen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "lightblue",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "margin": 14,
        }
      }
    >
      Welcome To The Second Feature About Screen
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 14,
        }
      }
    >
      Hello World
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Toggle Hello World Message
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`AboutScreen tests Goodbye message displays when helloWorld state is hello 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "lightblue",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "margin": 14,
        }
      }
    >
      Welcome To The Second Feature About Screen
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 14,
        }
      }
    >
      Goodbye World
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Toggle Hello World Message
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
